### PR TITLE
meme: fix errors

### PR DESCRIPTION
**Pull request by baberlevi**
Using libxslt and & libxml2 was causing the following error:
``
xsltutils.c: In function ‘xsltSaveResultToString’:
xsltutils.c:1758:26: error: dereferencing pointer to incomplete type ‘xmlBuf {aka struct _xmlBuf’}
``
Meme documentation recommends using their built in version instead if this happens. 
http://meme-suite.org/doc/install.html?man_type=web#prereq_libxml

Also added perl-xml-parser which when missing caused an error at runtime, and image-magick, which is optionally used by meme for image output.

This build now passes our internal test.

### CODE REVIEW

File: var/spack/repos/builtin/packages/meme/package.py [hunk 1]
```diff
@@ -39,17 +39,17 @@ class Meme(AutotoolsPackage):
...
     depends_on('python@2.7:', type=('build', 'run'))
     depends_on('mpi', when='+mpi')
+    depends_on('image-magick', type=('build', 'run'))
+    depends_on('perl-xml-parser', type=('build', 'run'))
 
     # disable mpi support
     def configure_args(self):
         spec = self.spec
-        args = []
+        args = ['--enable-build-libxml2 --enable-build-libxslt']
```
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/meme/package.py]: Shouldn't this be two strings, not one?

Comment (baberlevi) [reply var/spack/repos/builtin/packages/meme/package.py]: Yeah, probably should - fixed. Seems meme site is currently down, so the travis tests might fail.

**Comment by baberlevi** [comment]
Meme site seems fixed, and I merged devel in to kick off travis again after their issues. I think this is ready for merge.

**Comment by baberlevi** [comment]
I added the variant for image-magick.  

I'd like to start adding check_build tests to packages, but haven't tried yet due to both time & just not yet understanding how it would work. We have tests for a lot of packages, but most (like this one) require some input files - and I don't know where those would go. I assume we don't want fasta files and such cluttering up the repo. Suggestions ?

**Comment by baberlevi** [comment]
bumping for a merge, please

**Comment by scheibelp** [comment]
Do packages link against `meme`? Having meme build its own versions of packages is a potential source of conflicts (but not if nothing links against it).

It's possible that `meme` isn't finding the spack-built `libxml2`/`libxslt`, and when it attempts to build them the spack-injected references to the spack-built `libxml2` etc. are causing this error (because for example maybe `meme` includes a newer `libxslt` than spack.

It looks like you can set the XML2_CONFIG environment variable to point to the `xml2-config` binary provided by `libxml2`. That may help `meme` find Spack's `libxml2` if it wasn't finding it before.

**Comment by baberlevi** [comment]
I checked for dependents and did a grep through the packages, nothing seems to link against meme. I haven't had a chance to look at the libxml2 issue any deeper yet.

File: var/spack/repos/builtin/packages/meme/package.py [hunk 2]
```diff
@@ -37,19 +37,19 @@ class Meme(AutotoolsPackage):
...
     depends_on('python@2.7:', type=('build', 'run'))
     depends_on('mpi', when='+mpi')
+    depends_on('image-magick', type=('build', 'run'), when='+image-magick')
+    depends_on('perl-xml-parser', type=('build', 'run'))
 
-    # disable mpi support
     def configure_args(self):
         spec = self.spec
-        args = []
+        args = ['--enable-build-libxml2', '--enable-build-libxslt']
```
Comment (scheibelp) [inline var/spack/repos/builtin/packages/meme/package.py]: @baberlevi could you add a comment here referencing the PR thread (i.e. `have meme build its own versions of libxml2/libxslt, see #6736`)? In that case I'd be fine with merging this as-is.

**Comment by baberlevi** [comment]
@adamjstewart  - would you take a look at this one while you're on your spring break merge binge ? I think all the requested changes have been made

**Pull request merged**